url78d0cbe382_TeeAaTeeUu_SitsiPlaseeraus_tgz-pJ8-sitsiplaseeraus_MainJ8/src/Pisteyttaja/PaikanPisteet.java

```diff
@@ -0,0 +1,393 @@
+    private boolean avec;
+    private boolean puoliso;
+    private double pariPisteet;
+    private double sukupuoliPisteet;
+    private double yhteysPisteet;
+
+    /**
+     * Alustaa pisteiden laskurin käyttöön.
+     *
+     * @param paikka
+     * @param laskin
+     */
+    protected PaikanPisteet(Paikka paikka, Laskin laskin) {
+        this.paikka = paikka;
+        this.onYhteyksia = false;
+        this.laskin = laskin;
+    }
+
+    /**
+     * Palauttaa kaikki pisteet.
+     *
+     * @return Paikan pisteet.
+     */
+    protected double palautaPisteet() {
+        this.yhteydet = this.paikka.getSitsaaja().palautaYhteydet();
+
+        alustaLuvut();
+
+        pariPisteet = this.tarkistaAvecJaPuoliso();
+        sukupuoliPisteet = this.tarkistaYmparillaOlevienSukupuolet();
+
+        yhteysPisteet = kayKaikkiYhteydetLapi();
+
+        return pariPisteet + sukupuoliPisteet + yhteysPisteet;
+    }
+
+    /**
+     * Palauttaa mahdolliset avec ja puolisopisteet.
+     *
+     * @return avec-pisteet
+     */
+    public double getPariPisteet() {
+        return pariPisteet;
+    }
+
+    /**
+     * Palauttaa "tyttöpoika-järjestyksen" pisteet.
+     *
+     * @return
+     */
+    public double getSukupuoliPisteet() {
+        return sukupuoliPisteet;
+    }
+
+    /**
+     * Palauttaa yhteyksien pisteet, eli kuinka hyvin järjestetty tykkäysten ja
+     * pienien riitojen maksimoimiseksi.
+     *
+     * @return yhteyspisteet.
+     */
+    public double getYhteysPisteet() {
+        return yhteysPisteet;
+    }
+
+    /**
+     * Onko paikan sitsaajalla yhteyksiä muihin sitsaajiin.
+     *
+     * @return
+     */
+    protected boolean onkoYhteyksia() {
+        return this.onYhteyksia;
+    }
+
+    private double palautaPisteet(int arvo, int kohteidenErotus, int paikka) {
+        if (paikkaOnVasemmalla(paikka) == true) {
+            return pisteetKunPaikkaOnVasemmalla(kohteidenErotus, arvo);
+        } else {
+            return pisteetKunPaikkaOnOikealla(kohteidenErotus, arvo);
+        }
+
+    }
+
+    /**
+     * Palauttaa true, jos annettu paikka on vasemmalla puolella pöytää.
+     *
+     * @param paikka
+     * @return true, jos vasemmalla puolella.
+     */
+    public static boolean paikkaOnVasemmalla(int paikka) {
+        return paikka % 2 == 0;
+    }
+
+    private static boolean kohdeOnSamallaPuolella(int kohteidenErotus) {
+        return Math.abs(kohteidenErotus) % 2 == 0;
+    }
+
+    private double pisteetKunPaikkaOnVasemmalla(int kohteidenErotus, int arvo) {
+        if (kohdeOnSamallaPuolella(kohteidenErotus) == true) {
+            return 1.0 * arvo / Math.abs(kohteidenErotus / 2);
+        } else {
+            if (kohteidenErotus > 0) {
+                return 1.0 * arvo / this.laskin.hypot(1, 1.0 * (kohteidenErotus + 1) / 2);
+            } else {
+                return 1.0 * arvo / this.laskin.hypot(1, 1.0 * (Math.abs(kohteidenErotus) - 1) / 2);
+            }
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 118:126, 130:138.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.